x/mongo/driver/topology/connection.go

```diff
@@ -406,6 +408,18 @@ func (c *connection) read(ctx context.Context, dst []byte) (bytesRead []byte, er
 	// read the length as an int32
 	size := (int32(sizeBuf[0])) | (int32(sizeBuf[1]) << 8) | (int32(sizeBuf[2]) << 16) | (int32(sizeBuf[3]) << 24)
 
+	// In the case of an isMaster response where MaxMessageSize has not yet been set, use the hard-coded
+	// maxMessageSizeDefault instead.
+	maxMessageSize := c.desc.MaxMessageSize
+	if maxMessageSize == 0 {
+		maxMessageSize = maxMessageSizeDefault
+	}
+	if uint32(size) > maxMessageSize {
+		c.close()
+		err := errors.New("length of read message too large")
```

Comment:
nit: declare this as a constant at the top of the file (e.g. `var errResponseTooLarge = errors.New(...)`).

Reply:
Moved to constant `errResponseTooLarge`.